core/src/language/typescript.rs

```diff
@@ -219,12 +219,14 @@ impl TypeScript {
             .format_type(&field.ty, generic_types)
             .map_err(|e| std::io::Error::new(std::io::ErrorKind::Other, e))?;
         let optional = field.ty.is_optional() || field.has_default;
+        let double_optional = field.ty.is_double_optional();
         writeln!(
             w,
-            "\t{}{}: {};",
+            "\t{}{}: {}{};",
             typescript_property_aware_rename(&field.id.renamed),
             optional.then(|| "?").unwrap_or_default(),
-            ts_ty
+            ts_ty,
+            double_optional.then(|| " | null").unwrap_or_default()
```

Comment:
I'm not a typescript expert - is there a reason why this was chosen as `null` instead of `undefined`?

Reply:
Sure, there's actually a difference here. `undefined` usually means that a given field does not have a defined value, `null` specifically means that the field does have a value, but it's a `nothing` value.

In this context `null` is the right choice, since we specifically want to declare that we want to "zeroize/reset" the existing value during a partial update.

Reply:
Thanks for the explanation!